packages/server/modules/core/repositories/versions.ts

```diff
@@ -1,31 +1,26 @@
-import { Commits, knex, StreamCommits } from '@/modules/core/dbSchema'
+import { BranchCommits, knex, Branches, Commits } from '@/modules/core/dbSchema'
 import { Version } from '@/modules/core/domain/commits/types'
 import { Knex } from 'knex'
+import { groupBy } from 'lodash'
 
-const tables = {
-  versions: (db: Knex) => db<Version>(Commits.name)
-}
-
-export const getLastVersionByProjectIdFactory =
+export const getLastVersionsByProjectIdFactory =
   ({ db }: { db: Knex }) =>
   async ({
     projectIds
   }: {
     projectIds: readonly string[]
-  }): Promise<Record<string, Version & { projectId: string }>> => {
-    const results = await tables
-      .versions(db)
-      .join(StreamCommits.name, StreamCommits.col.commitId, Commits.col.id)
-      .whereIn(StreamCommits.col.streamId, projectIds)
-      .distinctOn(StreamCommits.col.streamId)
-      .select([...Commits.cols, knex.raw(`stream_commits."streamId" as "projectId"`)])
+  }): Promise<Record<string, Array<Version & { projectId: string }>>> => {
```

Comment:
`{ [something: string]: X}` is preferable to `Record<string, X>`, because it also explains the semantic meaning of the key

Reply:
also, its probably `Array | undefined` not `Array`. with this kind of type you're saying that any arbitrary string key will always have an array value